tests/tls_challenge_test.go

```diff
@@ -83,12 +84,18 @@ func (ts *TLSChallengeSuite) TestTLSChallenge() {
 	ts.mustSetDeploymentEnvVal(ts.ctx, s, sensorDeployment, sensorContainer, centralEndpointVar, proxyEndpoint)
 	ts.logf("Sensor will now attempt connecting via the nginx proxy.")
 
-	ts.waitUntilLog(ts.ctx, s, map[string]string{"app": "sensor"}, sensorContainer, "contain info about successful connection",
+	logMatchers := []logMatcher{
 		containsLineMatching(regexp.MustCompile("Info: Add central CA cert with CommonName: 'Custom Root'")),
-		containsLineMatching(regexp.MustCompile("Info: Connecting to Central server "+proxyEndpoint)),
 		containsLineMatching(regexp.MustCompile("Info: Established connection to Central.")),
 		containsLineMatching(regexp.MustCompile("Info: Communication with central started.")),
-	)
+	}
+	// The following logline was added after v3.74 and can't show up in the logs when running compatibility tests with
+	// sensor version 3.74 (support exception), see:
+	// https://github.com/stackrox/stackrox/blob/228c38c33033b20c1ff18b89650b9c137f4f7ff2/sensor/common/sensor/sensor.go#L167
+	if os.Getenv("SENSOR_3_74_SUPPORT_EXCEPTION") != "true" {
+		logMatchers = append(logMatchers, containsLineMatching(regexp.MustCompile("Info: Connecting to Central server "+proxyEndpoint)))
+	}
```

Comment:
Unfortunately, this is not correct. This line was introduced 5 years ago and 3.74 is roughly 2-2.5 years old. 
See: https://github.com/stackrox/stackrox/commit/3110efca9aa988ca3d9995e7b7b4c36b8d179c02#diff-860be736f75aa9a753048890d78329f39c7c7305f5186515d8159e0c902674adR110

Reply:
Oh, then I'll have to take another look. I only glanced at the git blame and saw that the line was modified in May 2024, but now that I looked at the commit for that I see that only a comment was added. In that case there must be some other reason why the test is failing, I'll investigate.